src/main/java/use_cases/death_detector_use_case/DeathDetectorInteractor.java

```diff
@@ -0,0 +1,91 @@
+package use_cases.death_detector_use_case;
+
+import entities.GameEvent;
+import entities.GameState;
+import entities.Player;
+import entities.cardEffects.CardEffect;
+import entities.cardEffects.CreatureStatsEffect;
+import entities.cards.CreatureCard;
+import entities.cards.StructureCard;
+import entities.cards.TargetType;
+import use_cases.GameOutputBoundary;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class DeathDetectorInteractor implements DeathDetectorInputBoundary {
+    final GameState gameState;
+    final DeathDetectorOutputBoundary deathDetectorPresenter;
+
+    public DeathDetectorInteractor(GameState gameState, DeathDetectorOutputBoundary deathDetectorPresenter) {
+        this.gameState = gameState;
+        this.deathDetectorPresenter = deathDetectorPresenter;
+    }
+
+    private void processPlayer(Player player1, Player player2, List<Integer> ids) {
+        List<Integer> defeatCreatureIds = new ArrayList<>();
+        for (CreatureCard c : player1.getCreatures()) {
+            if (c.getHitPoints() <= 0) {
+                defeatCreatureIds.add(c.getId());
+                player1.removeCreature(c.getId());
+            }
+        }
```

Comment:
@ethanhsu8004 raised a good point, I also commented on the other pull request with the same problem. His solution is to use an iterator, but this is more readable.

```suggestion
        List<CreatureCard> deadCreatures = new ArrayList<>();
        for (CreatureCard c : player1.getCreatures()) {
            if (c.getHitPoints() <= 0) {
                defeatCreatureIds.add(c.getId());
                deadCreatures.add(c);
            }
        }
        player1.getCreatures().removeAll(deadCreatures);
```

Reply:
nb got it